projects/components/src/greeting-label/greeting-label.component.ts

```diff
@@ -0,0 +1,40 @@
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { TypedSimpleChanges } from '@hypertrace/common';
+
+@Component({
+  selector: 'htc-greeting-label',
+  styleUrls: ['./greeting-label.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: ` <div class="greeting-label">{{ this.greetingMessage }}</div>`
+})
+export class GreetingLabelComponent implements OnChanges {
+  @Input()
+  public readonly label: string = '';
+
+  public greetingMessage: string = '';
+
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.label) {
+      this.setGreetingMessage();
+    }
+  }
+
+  private setGreetingMessage(): void {
+    const parts = this.label.split('{greeting}');
```

Comment:
I don't love this - it means the caller needs to know the implementation to know how to follow the specific template rules of this component. Do we ever envision this being used where the greeting comes anywhere but the beginning of the message? If not, then maybe we just remove the templating from the label message itself. Because I could also argue that I could use `{greeting}` twice in my message.

Reply:
As of now, we don't have a use case for it to be anywhere other than the beginning. But would the flexibility hurt? In case we decide to use it at other locations in the future, this could be handy.
As for the possibility of using `{greeting}` twice in the message, we can obviously add a check against it, but that'd be an overkill IMO. I think its ok to leave that upto the user to use it correctly.

Reply:
IMO both points go together.
> we don't have a use case for it to be anywhere other than the beginning. But would the flexibility hurt?

We're providing flexibility for one scenario but not another, and neither are needed. Generally, we want the API to be as simple as we can to support the use cases we have. When adding more functionality/flexibility, then it should be clear and consistent (not going to lie and try to argue we're always the best about this). Here, I feel like it makes the API more confusing (because now as a caller you need to understand how to formulate a template string - does the widget include the comma or do I?! what's the template var name? Should I capitalize?) and more non-standard/unexpected (because unlike a standard template string, you can only reference the template var once)

Reply:
Assuming we don't want the greeting to be templatized, the problem about understanding the API still persists.
example: if I want to show `Good morning, here is your report`, assuming that greeting is always in the beginning, the consumer still needs to know if the user supplied template has to be with or without the comma.
So, not templatizing the label doesn't solve that problem.

For now, I'll make this simple and go with prefixing the greeting.

Reply:
> assuming that greeting is always in the beginning, the consumer still needs to know if the user supplied template has to be with or without the comma.
So, not templatizing the label doesn't solve that problem.

Abstractly, I think of it as just a message that is provided. The widget could choose to use a common, a period and new line, whatever - the widget's concern is to put the provided message in a greeting context. So I guess I'd think the comma belongs to the greeting too, and the message can be just `here is your report` (the greeting widget handling capitalization as appropriate). Anyway, this is such a strange and special cased widget, that we probably don't have to worry too much about this.